server/plugin/taskLinkPreview_test.go

```diff
@@ -1,58 +1,129 @@
 package plugin
 
 import (
+	"errors"
 	"net/http"
 	"testing"
 
 	"github.com/golang/mock/gomock"
+	"github.com/mattermost/mattermost-server/v5/plugin/plugintest"
 	"github.com/stretchr/testify/assert"
 
 	"github.com/mattermost/mattermost-plugin-azure-devops/mocks"
 	"github.com/mattermost/mattermost-plugin-azure-devops/server/serializers"
+	"github.com/mattermost/mattermost-plugin-azure-devops/server/testutils"
 )
 
 func TestPostTaskPreview(t *testing.T) {
-	p := Plugin{}
+	mockAPI := &plugintest.API{}
 	mockCtrl := gomock.NewController(t)
 	mockedClient := mocks.NewMockClient(mockCtrl)
-	p.Client = mockedClient
+	p := setupMockPlugin(mockAPI, nil, mockedClient)
 	for _, testCase := range []struct {
 		description string
 		linkData    []string
+		err         error
+		statusCode  int
 	}{
 		{
 			description: "PostTaskPreview: valid",
-			linkData:    []string{"https:", "", "dev.azure.com", "abc", "xyz", "_workitems", "edit", "1"},
+			linkData:    []string{"https:", "", "test.com", "abc", "xyz", "_workitems", "edit", "1"},
+			statusCode:  http.StatusOK,
+		},
+		{
+			description: "PostTaskPreview: invalid",
+			linkData:    []string{"https:", "", "test.com", "abc", "xyz", "_workitems", "edit", "1"},
+			err:         errors.New("failed to post task preview"),
+			statusCode:  http.StatusInternalServerError,
 		},
 	} {
 		t.Run(testCase.description, func(t *testing.T) {
-			mockedClient.EXPECT().GetTask(gomock.Any(), gomock.Any(), gomock.Any(), "mockUserID").Return(&serializers.TaskValue{}, http.StatusOK, nil)
-			resp, err := p.PostTaskPreview(testCase.linkData, "mockUserID", "mockChannelID")
-			assert.Equal(t, "", err)
-			assert.NotNil(t, resp)
+			mockAPI.On("LogDebug", testutils.GetMockArgumentsWithType("string", 3)...)
+			mockedClient.EXPECT().GetTask(gomock.Any(), gomock.Any(), gomock.Any(), mockMattermostUserID).Return(&serializers.TaskValue{}, testCase.statusCode, testCase.err)
+
+			resp, msg := p.PostTaskPreview(testCase.linkData, mockMattermostUserID, mockChannelID)
+			assert.Equal(t, "", msg)
+			if testCase.err != nil {
+				assert.Nil(t, resp)
```

Comment:
can we check the response here as well if possible

Reply:
Not possible as the response is returned by `model.ParseSlackAttachment`